client/src/views/SearchResults.tsx

```diff
@@ -0,0 +1,256 @@
+    display: 'flex',
+    flexDirection: 'column',
+    justifyContent: 'flex-start',
+    padding: '10px 5%',
+    width: '80%',
+  },
+  makeAPost: {
+    fontWeight: 'bold',
+    textDecoration: 'none',
+    color: 'black',
+  },
+}));
+
+function SearchResults(): JSX.Element {
+  const classes = useStyles();
+  const history = useHistory();
+
+  const { search: querySearchText, category: querySearchCategory } = queryString.parse(
+    history.location.search,
+  );
+
+  const [searchCategory, setSearchCategory] = React.useState<string>(
+    (querySearchCategory as string) || '',
+  );
+  const [searchText, setSearchText] = React.useState<string>((querySearchText as string) || '');
+  const [inputText, setInputText] = React.useState<string>('');
+  const [selectedFilters, setSelectedFilters] = React.useState<{ [key: string]: boolean }>({});
+
+  const handleCheck = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setSelectedFilters({
+      ...selectedFilters,
+      [event.target.value]: event.target.checked,
+    });
+  };
+
+  const [offers, setOffers] = React.useState<Asset[]>([]);
+  const [needs, setNeeds] = React.useState<Asset[]>([]);
+  const [orgs, setOrgs] = React.useState<Organization[]>([]);
+  //TODO: create Volunteer Type in types/index.ts and change Object[] to Volunteer[] below
+  const [volunteer, setVolunteer] = React.useState<Object[]>([]);
+
+  const handleChange = (event: SelectChangeEvent) => {
+    setSearchCategory(event.target.value);
+  };
+
+  const handleSearch = () => {
+    setSearchText(inputText);
+    setInputText('');
+  };
+
+  const handleInput = (e: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => {
+    setInputText(e.target.value);
+  };
+
+  React.useEffect(() => {
+    if (searchCategory === 'Volunteer') {
+      //TODO: Change API to /volunteer
+      fetch(`${APP_API_BASE_URL}/organizations`)
+        .then((resp) => resp.json())
+        .then((data: Asset[]) => {
+          setVolunteer(data);
+        });
+    } else if (searchCategory === 'Offers' || searchCategory === 'Needs') {
+      fetch(
+        `${APP_API_BASE_URL}/assets?type=${searchCategory === 'Needs' ? 'donation' : 'request'}${
+          searchText ? `&title=${searchText}` : ''
+        }`,
+      )
+        .then((resp) => resp.json())
+        .then((data: Asset[]) => {
+          if (searchCategory === 'Needs') {
+            setNeeds(data);
+          } else {
+            setOffers(data);
+          }
+        });
+    } else if (searchCategory === 'Nonprofits') {
+      fetch(`${APP_API_BASE_URL}/organizations?search=${searchText}`)
+        .then((resp) => resp.json())
+        .then((data: Organization[]) => {
+          setOrgs(data);
+        });
+    } else {
+      //TODO: Change API to fetch ALL combined data
+      console.log('Fetches "All Lists" API here');
+    }
+
+    history.push(`/SearchResults?search=${searchText}&category=${searchCategory}`);
+  }, [history, searchText, searchCategory]);
```

Comment:
@jd2rogers2 Agreed, two states created confusion - my only intention was not to make API calls at every onKeyPress.

The original file did not consider the URL query; thus, the `querySearchText` variable was never updated. I've tried using the old file, and it did not work as intended.

Additionally, I don't believe `querySearchText` is correct for line 147 above because this does not change the query in the URL and thus does not update the page after the fetch. `querySearchText` is constant and is taken from the previous home page search and is not a state that changes. `searchText` does change. Therefore, I chose it to go into the history.push on "ENTER" or on Search Icon Click.

I believe this iteration, I made sure to take into account your initial intentions. 
Please review this commit with the original intentions of:
- Textfield shows the searched word/phrase but does not fire on every keypress
- Updating once the search is pressed
- Keeping URL query update with the latest search
- All other variables are updated once the search is made